test/core/util/fake_stats_plugin.h

```diff
@@ -64,7 +65,9 @@ class FakeClientCallTracer : public ClientCallTracer {
     explicit FakeClientCallAttemptTracer(
         std::vector<std::string>* annotation_logger)
         : annotation_logger_(annotation_logger) {}
-    ~FakeClientCallAttemptTracer() override {}
+    ~FakeClientCallAttemptTracer() override {
```

Comment:
Using a notification wait in a destructor seems scary.. Instead, why not just allocate it on the heap and delete it on `RecordEnd`?

Reply:
I don't think there's necessarily anything wrong with waiting on the notification here, but I agree that it seems cleaner to just ensure the object isn't destroyed until `RecordEnd()` is called.  Note though that we can't just unconditionally destroy the object in `RecordEnd()`, because when we're not yet at the end of the test, we need the object to still exist so that we can access it in the test code.  So I think we'll want to make `FakeClientCallAttemptTracer` be ref-counted, take a new ref to store in `FakeClientCallTracer::call_attempt_tracers_`, and then have `RecordEnd()` drop the original ref.

Reply:
Thanks! Done. PTAL.

Reply:
That makes sense.